buildenvs/base.Dockerfile

```diff
@@ -2,114 +2,79 @@
-# COPY --from=qemu /lib/arm-linux-gnueabihf/ /lib/arm-linux-gnueabihf
-# COPY --from=qemu /lib/aarch64-linux-gnu/ /lib/aarch64-linux-gnu
-COPY --from=kraftkit /kraft /usr/local/bin
-
-ARG GCC_PREFIX=x86_64-linux-gnu
-
-# Link the GCC toolchain
-RUN set -xe; \
-    ln -s /bin/${GCC_PREFIX}-as         /bin/as; \
-    ln -s /bin/${GCC_PREFIX}-ar         /bin/ar; \
-    ln -s /bin/${GCC_PREFIX}-c++        /bin/c++; \
-    ln -s /bin/${GCC_PREFIX}-c++filt    /bin/c++filt; \
-    ln -s /bin/${GCC_PREFIX}-elfedit    /bin/elfedit; \
-    ln -s /bin/${GCC_PREFIX}-gcc        /bin/cc; \
-    ln -s /bin/${GCC_PREFIX}-gcc        /bin/gcc; \
-    ln -s /bin/${GCC_PREFIX}-gcc-ar     /bin/gcc-ar; \
-    ln -s /bin/${GCC_PREFIX}-gcc-nm     /bin/gcc-nm; \
-    ln -s /bin/${GCC_PREFIX}-gcc-ranlib /bin/gcc-ranlib; \
-    ln -s /bin/${GCC_PREFIX}-gccgo      /bin/gccgo; \
-    ln -s /bin/${GCC_PREFIX}-gcov       /bin/gcov; \
-    ln -s /bin/${GCC_PREFIX}-gcov-dump  /bin/gcov-dump; \
-    ln -s /bin/${GCC_PREFIX}-gcov-tool  /bin/gcov-tool; \
-    ln -s /bin/${GCC_PREFIX}-gprof      /bin/gprof; \
-    ln -s /bin/${GCC_PREFIX}-ld         /bin/ld; \
-    ln -s /bin/${GCC_PREFIX}-nm         /bin/nm; \
-    ln -s /bin/${GCC_PREFIX}-objcopy    /bin/objcopy; \
-    ln -s /bin/${GCC_PREFIX}-objdump    /bin/objdump; \
-    ln -s /bin/${GCC_PREFIX}-ranlib     /bin/ranlib; \
-    ln -s /bin/${GCC_PREFIX}-readelf    /bin/readelf; \
-    ln -s /bin/${GCC_PREFIX}-size       /bin/size; \
-    ln -s /bin/${GCC_PREFIX}-strings    /bin/strings; \
-    ln -s /bin/${GCC_PREFIX}-strip      /bin/strip;
-
-ENV LC_ALL=C.UTF-8
-ENV LANG=C.UTF-8
-ENV KRAFTKIT_LOG_TYPE=basic
-ENV DOCKER=
+COPY --from=qemu     /bin/        /usr/local/bin
+COPY --from=qemu     /share/qemu/ /share/qemu
+COPY --from=kraftkit /kraft       /usr/local/bin
 
 # Install unikraft dependencies
 RUN set -xe; \
     apt-get update; \
     apt-get install -y --no-install-recommends \
-      make=4.3-4.1 \
-      libncursesw5-dev=6.2+20201114-2+deb11u1 \
-      libncursesw5=6.2+20201114-2+deb11u1 \
-      libyaml-dev=0.2.2-1 \
-      flex=2.6.4-8 \
-      git=1:2.30.2-1+deb11u2 \
-      wget=1.21-1+deb11u1 \
-      patch=2.7.6-7 \
-      gawk=1:5.1.0-1 \
-      socat=1.7.4.1-3 \
-      bison=2:3.7.5+dfsg-1 \
-      bindgen=0.55.1-3+b1 \
-      bzip2=1.0.8-4 \
-      unzip=6.0-26+deb11u1 \
-      uuid-runtime=2.36.1-8+deb11u1 \
-      openssh-client=1:8.4p1-5+deb11u1 \
-      autoconf=2.69-14 \
-      xz-utils=5.2.5-2.1~deb11u1 \
-      python3=3.9.2-3 \
-      ca-certificates=20210119; \
+      autoconf \
+      bindgen \
+      bison \
+      bzip2 \
+      ca-certificates \
+      flex \
+      gawk \
+      gcc-12 \
+      gcc-12-aarch64-linux-gnu \
+      gcc-12-arm-linux-gnueabihf \
+      git \
+      libncursesw5 \
+      libncursesw5-dev \
+      make \
+      openssh-client \
+      patch \
+      python3 \
+      socat \
+      unzip \
+      uuid-runtime \
+      wget \
+      xz-utils \
+      ; \
```

Comment:
Don't forget to also pin the versions so we make sure nothing changes

Reply:
I specifically removed pinning as we want to be able to easily upgrade dependencies simply by re-running the container build. This makes it easier from a maintenance point of view.

Reply:
Hmm, sure, but if something were to break, we won't know from sure what package caused it. It's fine from me, but Codacy was complaining a while ago

Reply:
Don't forget to also add `curl` while at it, doesn't hurt to have it in

Reply:
I can understand why Codacy might complain but until there's a dependabot workflow that can specifically upgrade this, then this is the easier way to move forward.

I can add `curl`.